### ISSUE TITLE
dts-generator ignores child namespaces/enums etc. of classes

### ISSUE THREAD

**Issue by codeworrior** [SAP/ui5-typescript#48]
In UI5, classes can have namespaces, classes, enums, interfaces or typedefs as static children.
Examples:
* [sap.ui.core.Popup.Dock](https://openui5nightly.hana.ondemand.com/api/sap.ui.core.Popup.Dock) (namespace, rather should be an enum - never mind)
* [sap.ui.core.View.Preprocessor](https://openui5nightly.hana.ondemand.com/api/sap.ui.core.mvc.View.Preprocessor) (interface)
* [sap.ui.core.XMLView.PreprocessorType](https://openui5nightly.hana.ondemand.com/api/sap.ui.core.mvc.XMLView.PreprocessorType) (enum)
* [sap.ui.core.UIComponent.RoutingMetadata](https://openui5nightly.hana.ondemand.com/api/sap.ui.core.UIComponent.RoutingMetadata) (type definition)

The dts-generator currently ignores such children. It assumes, that only namespaces can contain namespaces, classes etc. When such types are referenced in signatures, they are reported as unresolved dotted type names and replaced with type `any`. 

According to the following [stackoverflow question](https://stackoverflow.com/questions/29844959/enum-inside-class-typescript-definition-file),  it should be possible to represent the class as both, a `class` and a `namespace` in such cases. All nested namespaces, classes, enums etc. should go into the `namespace` declaration, all instance methods and fields as well as simple static members (functions, fields) should go into the `class` definition.

I tried the following in VS code (patching the sap.ui.core.d.ts) and it worked for me
```js
  namespace Popup {
    enum Dock { ... }
  }
  class Popup {
    open()
    close()
  }
```

**Comment by bd82** [comment]
Closing for tasks cleanup, may be re-opened / created as we discuss on future tasks for this project.
